help.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"fmt"
+	"html/template"
+	"io"
+	"os"
+	"strings"
+)
+
+// runHelp implements the 'help' command.
+func runHelp(args []string) int {
+	if len(args) == 0 {
+		mainUsage(os.Stdout)
+		return 0
+	}
+	if len(args) != 1 {
+		fmt.Fprintf(os.Stderr, "modcop help %s: too many arguments\n", strings.Join(args, " "))
+		return 2
+	}
+	t := template.Must(template.New("").Parse(commandHelpTemplate))
+	for _, c := range commands {
+		if c.Name() == args[0] {
+			if err := t.Execute(os.Stdout, c); err != nil {
+				errorf("cannot write usage output: %v", err)
+			}
+			return 0
+		}
+	}
+	fmt.Fprintf(os.Stderr, "gohack help %s: unknown command\n", args[0])
+	return 2
+}
+
+func mainUsage(f io.Writer) {
+	t := template.Must(template.New("").Parse(mainHelpTemplate))
+	if err := t.Execute(f, commands); err != nil {
+		errorf("cannot write usage output: %v", err)
+	}
+}
+
+var mainHelpTemplate = `
+The modcop command checks that module dependencies do not stray outside
```

Comment:
I've perhaps missed a few beats since we last discussed this, but doesn't `modcop` operate at a number of different levels?

* org-wide
* module
* package

i.e. constraints can be described at any/all levels?

Reply:
I changed "module" to "package".